lib/api/apiUtils/authorization/prepareRequestContexts.js

```diff
@@ -193,13 +194,28 @@ function prepareRequestContexts(apiMethod, request, sourceBucket,
                 const putObjectLockRequestContext =
                   generateRequestContext('objectPutRetention');
                 requestContexts.push(putObjectLockRequestContext);
+                if (hasGovernanceBypassHeader(request.headers)) {
```

Comment:
in the previous code, there was an extra condition `isRequesterNonAccountUser(authInfo)` : which skips the user-governance-bypass right for "non-account user"...

is this not needed anymore? Will this skipping of the `bypassGovernanceRetention` check be done automatically by Vault?

Reply:
Yes, as explained in the commit message, when we authorize against vault, and the identity is an account, the result is an Allow, and no policy is evaluated. So in all cases, we send this action to be authorized:
- If account: allowed, the overhead is that we added one more action to check, but we do that in a single api call instead of 2 sequential calls.
- if not account: same as before, but in a single api call.